### DISCUSSION TITLE
charge_history query parameters

### DISCUSSION THREAD

**Discussion by MCRE-BE** [timdorr/tesla-api#788] [Q&A] [answered]
Hi all,

I have been playing with `charge_history` for my Tesla model 3. My goal is to be able to pull from the API the totals per month of #KWH charged per type of charging location. This data is accessible through the API, by default, for 31 days. This exactly the app behaviour.

However, in the app you can also scroll from month to month. I am wondering if anybody knows how they do this from an API perspective. I am wondering if something like this is possible? For chargers there is this possibility, should this work also for vehicules?

https://github.com/Teslemetry/node-tesla-fleet-api/blob/359cf9c635505b9c699cd689666e712623564c7d/src/energy.ts#L76C2-L89C4

Thanks

**Comment by brodikAE** [comment, marked as answer]
something like
`api/1/vehicles/<YOUR_VIN>/charge_history?country_label=<YOUR_COUNTRY>&currency_code=<YOUR_CURRENCY>&start_date=2024-08-01T00%3A00%3A00%2B02%3A00&client_time_zone=<YOUR_TIMEZONE>&client_country=<YOUR_COUNTRY>&end_date=2024-08-31T23%3A59%3A59%2B02%3A00&interval=day ...`

**Reply by MCRE-BE** [reply]
Thanks !  I'll try.

**Reply by MCRE-BE** [reply]
@brodikAE : That does seem to work ! I'll see if I can add the information to some documentation somewhere as I didn't manage to find it online.

Sample code that worked : 
```python
name = "ChargeHistory"
endpoint = {
    "TYPE": "POST",
    "URI": f"api/1/vehicles/{hist.vehicle["id"]}/charge_history",
    "AUTH": True,
}
uri = endpoint["URI"]
serialize = endpoint.get("CONTENT") != "HTML" and name != "STATUS"
hist.vehicle.tesla.request(
    "POST",
    uri,
    serialize,
    withhold_token=not endpoint["AUTH"],
    params={
        "country_label": "FR",
        "currency_code": "EUR",
        "vehicle_id": hist.vehicle["id"],
        "start_date": "2024-07-01T00:00:00-08:00",
        "end_date": "2024-09-01T00:00:00-08:00",
        "client_time_zone": "Europe/Brussels",
        "client_country": "FR",
        "interval": "day",
    },
)
```

<img width="781" alt="Capture d’écran 2024-11-07 à 17 31 18" src="https://github.com/user-attachments/assets/9ac0e58b-52b8-4126-8d19-add6badd9273">
